src/main/java/com/limechain/runtime/hostapi/HostApi.java

```diff
@@ -50,69 +53,77 @@ protected static ImportObject getImportObject(final String functionName,
         }, args, EMPTY_LIST_OF_TYPES);
     }
 
+    @Deprecated(forRemoval = true)
     public static byte[] getDataFromMemory(long pointer) {
-        int ptr = (int) pointer;
-        int ptrLength = (int) (pointer >> 32);
-        byte[] data = new byte[ptrLength];
-
-        Memory memory = getMemory();
-        memory.buffer().get(ptr, data, 0, ptrLength);
-        return data;
+        return null;
     }
 
+    @Deprecated(forRemoval = true)
     public static int putDataToMemory(byte[] data) {
-        Memory memory = getMemory();
-        ByteBuffer buffer = getByteBuffer(memory);
-        int position = buffer.position();
-        buffer.put(position, data, 0, data.length);
-        buffer.position(position + data.length);
-
-        return position;
-    }
-
-    protected static Memory getMemory() {
-        return runtime.getInstance().exports.getMemory("memory");
-    }
-
-    public static void setRuntime(Runtime runtime) {
-        HostApi.runtime = runtime;
-    }
-
-    public static void setRepository(KVRepository<String, Object> repository) {
-        HostApi.repository = repository;
+        return -1;
```

Comment:
Maybe until removal provide a fallback implementation to the deprecated method

```suggestion
        return writeDataToMemory(data).pointer();
```